### PR TITLE
HTBHF-1279 Refactored some of the new card service logic into a new c…

### PR DESCRIPTION
**Pull request by dwybourn**
…lass.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/eligibility/EligibilityResponse.java [hunk 1]
```diff
@@ -24,4 +24,10 @@
 
     @JsonProperty("children")
     private final List<ChildDTO> children;
+
+    public static EligibilityResponse withStatus(EligibilityStatus eligibilityStatus) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/eligibility/EligibilityResponse.java]: Can we call this `buildWithStatus` please?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityService.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+@Service
+@AllArgsConstructor
+public class EligibilityService {
+
+    private final EligibilityClient client;
+    private final EligibilityStatusCalculator eligibilityStatusCalculator;
+    private final ClaimRepository claimRepository;
+
+    public EligibilityResponse determineEligibility(Claimant claimant) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityService.java]: JavaDoc?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    @Test
+    void shouldReturnDuplicateResponseWhenLiveClaimAlreadyExists() {
+        given(claimRepository.liveClaimExistsForNino(any())).willReturn(true);
+        Claimant claimant = aValidClaimant();
+
+        EligibilityResponse eligibilityResponse = eligibilityService.determineEligibility(claimant);
+
+        assertThat(eligibilityResponse.getEligibilityStatus()).isEqualTo(DUPLICATE);
+        verify(claimRepository).liveClaimExistsForNino(claimant.getNino());
+        verifyZeroInteractions(client);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityServiceTest.java]: `verifyZeroInteractions` takes a varargs, so you can use `verifyZeroInteractions(client, eligibilityStatusCalculator)`.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/eligibility/EligibilityResponse.java (hunk 1 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/eligibility/EligibilityResponse.java]: Done, and replaced code with static import.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityService.java (hunk 2 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityService.java]: Added.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityServiceTest.java (hunk 3 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityServiceTest.java]: Done.

**Pull request merged**